src/interceptors/WebSocket/WebSocketClientConnection.ts

```diff
@@ -5,44 +5,53 @@
  * meant to be used over any WebSocket implementation
  * (not all of them follow the one from WHATWG).
  */
-import type { WebSocketRawData, WebSocketTransport } from './WebSocketTransport'
+import type { WebSocketData, WebSocketTransport } from './WebSocketTransport'
 import { WebSocketMessageListener } from './WebSocketOverride'
 import { bindEvent } from './utils/bindEvent'
 import { CloseEvent } from './utils/events'
 import { uuidv4 } from '../../utils/uuid'
 
 const kEmitter = Symbol('kEmitter')
 
+export interface WebSocketClientConnectionProtocol {
+  id: string
+  url: URL
+  send(data: WebSocketData): void
+  close(code?: number, reason?: string): void
+}
+
 /**
  * The WebSocket client instance represents an incoming
  * client connection. The user can control the connection,
  * send and receive events.
  */
-export class WebSocketClientConnection {
+export class WebSocketClientConnection
```

Comment:
Extending this class itself to implement connection-like objects is inefficient. It requires to provide a bunch of irrelevant properties, like `kEmitter`, `socket`, and `transport`. Those are internal details of this class but extending it means providing those internal details also.